src/transports/tcp.rs

```diff
@@ -410,32 +419,58 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
     }
 
     fn send(endpoint: &mut Self::Endpoint, data: &[u8]) -> PeerNetResult<()> {
+        let start_time = Instant::now();
         let msg_size: u32 = data.len().try_into().map_err(|_| {
             TcpError::ConnectionError
                 .wrap()
                 .error("send len too long", Some(format!("{:?}", data.len())))
         })?;
-        if msg_size > 1048576000 {
+        if msg_size > endpoint.config.max_message_size as u32 {
             return Err(TcpError::ConnectionError
                 .wrap()
-                .error("send len too long", Some(format!("{:?}", data.len()))))?;
+                .error("send len too long", Some(format!("{:?}", data.len()))));
+        }
+        // write msg len to stream
+        match endpoint.stream.write(&msg_size.to_be_bytes()) {
```

Comment:
Here as well we should use `write_exact_timeout` (see comment https://github.com/massalabs/PeerNet/pull/59/files#r1241286372 ) because write can sometimes return `Ok(1)` !